### PR TITLE
Initial commit converting customers actor to sqldb interface

### PR DESCRIPTION
**Pull request by autodidaddict**
Using the customers actor as a prototype. Once we're happy with how that one works, I'll repeat the pattern for the other two actors.

Would appreciate some help getting a test harness/docker compose set up here as I don't currently have a functioning docker _or_ postgres environment.

### CODE REVIEW

**Review by brooksmtownsend — APPROVED** [review]
I think this looks good, the SQL syntax is pretty clean 🧼

File: petclinic/actors/customers/Makefile [hunk 1]
```diff
@@ -13,3 +13,6 @@ CALL_ALIAS = petclinic/customers
 
 include ./actor.mk
 
+link-sqldb:
+	wash ctl link put $(shell make --silent actor_id) \
+	$(SQLDB_ID) $(shell base64 -w0 sql-linkdef.json)
```
Comment (stevelr) [inline petclinic/actors/customers/Makefile:18]: I know I used this in the todo example. I think you'll need the last part to be
`config_b64=$(shell base64 -w0 sql-linkdef.json)`

File: petclinic/actors/customers/actor.mk [hunk 2]
```diff
@@ -57,6 +57,7 @@ all:: $(DIST_WASM)
 $(DIST_WASM): $(UNSIGNED_WASM) Makefile
```
Comment (stevelr) [inline petclinic/actors/customers/actor.mk:57]: you shouldn't need an actor.mk in this folder. From your main folder, 
`include ../../../../build/makesilfes/actor.mk`
(with the appropriate number of `../`)

File: petclinic/actors/customers/src/lib.rs [hunk 3]
```diff
@@ -1,118 +1,96 @@
...
     }
 
-    async fn update_pet(&self, _ctx: &Context, _arg: &Pet) -> RpcResult<bool> {
-        Ok(true)
+    async fn update_pet(&self, ctx: &Context, arg: &Pet) -> RpcResult<bool> {
+        let db = SqlDbSender::new();
+        Ok(match db::update_pet(&ctx, &db, arg).await {
+            Ok(_) => true,
+            Err(_e) => false,
+        })
```
Comment (stevelr) [inline petclinic/actors/customers/src/lib.rs:81]: If you want to turn a result into a boolean, you can just use is_ok(), so
 `Ok(db::update_pet(&ctx,&db,arg).await.is_ok())`

**Review by stevelr — APPROVED** [review]

File: petclinic/actors/customers/src/lib.rs (hunk 3 above)
Comment (autodidaddict) [reply petclinic/actors/customers/src/lib.rs:81]: Yeah I left the match because I'm likely going to drop a call to error! in the fail arm.

**Pull request merged**